src/lib/reader.ts

```diff
@@ -78,9 +78,24 @@ class Reader {
   }
 
   /**
-   * Read a fixed-length unsigned big-endian integer.
+   * Read a fixed-length unsigned big-endian integer as a JS number.
    *
-   * @param {number} length Length of the integer in bytes.
+   * @param length Length of the integer in bytes.
+   */
+  readUIntNumber (length: number): number {
+    if (length >= 1 && length <= MAX_SAFE_BYTES) {
```

Comment:
Is there such thing as a 0-length number?

Reply:
No, zero-length numbers are not allowed. Both the previous and current versions of this library throw in that case and there is a test for it.